### DISCUSSION TITLE
Selective pass-through of some directories

### DISCUSSION THREAD

**Discussion by kevinopensquiggly** [statiqdev/Discussions#153] [Q&A]
I'm very new to Statiq and this may be an extremely basic question, but ...

I'm generating a static site which will be hosted in Azure Static Web Apps and have some embedded functions in them. To do that, I will add an api folder to my input folder, and place my function and other json files in that folder, per the Azure Static Web Apps instructions.

My input folder would then look like:

**input**
-   api  -> .js and .json files for Azure Static Web Apps functions
-   assets -> image files, etc.
-   contents -> all the content for my web site
-   css -> css files

The api folder needs to get passed through to the output unmodified. So the output should look like:

**output**
- api -> Direct pass through of input folder, all .js and .json files left unmodified
- assets -> image files passed through unmodified (works by default)
- contents -> processed/transformed by Statiq as per usual
- css -> css files passed through unmodified (works by default)

So I don't know how to do that. I don't know how to tell Statiq that the api folder is special and should be passed through, while the other folders should be processed as usual. And also, since the api folder contains json files, Statiq wants to ignore them and not pass them through.

I'm sure this is a bread-and-butter use case for Statiq, but even a simple use case as this does not jump out at me as to how to do it from reading the documentation.

My Program.cs file looks like this
```
public class Program
{
  public static async Task<int> Main(string[] aArgumentArray) =>
    await Bootstrapper
      .Factory
      .CreateDocs(aArgumentArray)
      .AddShortcode(/* excluded for brevity */)
      .AddShortcode(/* excluded for brevity */)
      .RunAsync();
}
```
I imagine what I'm supposed to do is add some pipeline processing into my Program.cs, but how to do that escapes me and the documentation is confusing.

Thank you.

**Comment by daveaglick** [comment]
Thanks for this detailed write-up!

What you need to do is set the `ContentType` and `MediaType` of the entire `api` directory to `Asset` (for the former, which tells Statiq these files are mostly to pass though) and `text/plain` (for the latter so it doesn't try to process SASS files or anything).

Setting these values for everything in the `api` directory can be done using [Directory Metadata](https://www.statiq.dev/guide/web/directory-metadata). This can be accomplished by creating a `_directory.yaml` file in your `api` directory with the following content:

```
ContentType: Asset
MediaType: text/plain
```

The process is described near the bottom of this page: https://www.statiq.dev/guide/web/content-and-media-types.

Let me know if this works for you!

> the documentation is confusing

That's extremely valuable feedback. While the steps above should do what you need, I agree it's not at all clear how to go from the use case "output everything in this folder unchanged" to those steps in the docs. I wonder if a "Common Use Cases" section or something would help, and then it could grow over time as more of these kinds of questions come up.
